go/private/extensions.bzl

```diff
@@ -201,6 +215,40 @@ def _go_sdk_impl(ctx):
                 sdk_version = wrap_tag.version,
             ))
 
+        # If the module suggests to read the toolchain version from a `go.mod` file, use that.
+        for index, from_file_tag in enumerate(module.tags.from_file):
```

Comment:
I may be missing something, but does this really end up registering multiple SDKs of the given version for the common exec platforms? A technique used in the `go_deps` extensions is to synthesize a tag (in this case `download`) as a struct and then process it just like the other download tags below.

Reply:
I didn't understand what you meant when you suggested synthesis earlier, but now I think I do.  In gazelles tags, it creates new lists of additional tags and then processes them together with the "real" tags. I will try this approach soon.

Reply:
Take a look at this please.